sequins.go

```diff
@@ -159,11 +161,52 @@ func (s *sequins) initCluster() error {
 	routableIpAddress := net.JoinHostPort(ip, port)
 	routableAddress := net.JoinHostPort(hostname, port)
 	shardID := s.config.Sharding.ShardID
+
+	var lock *gozk.Lock
 	if shardID == "" {
-		shardID = routableAddress
+		fp := filepath.Join(s.config.LocalStore, "self_assigned_id")
+		_, err := os.Stat(fp)
+		if err != nil {
+			if os.IsNotExist(err) {
+				log.Print("No self-assigned ID file found")
+
+				lock := zkWatcher.CreateIDAssignmentLock()
+				lock.Lock()
```

Comment:
Could we structure this in a way that we could add `defer lock.Unlock()`? Maybe move lines 165-209 to a separate function?